### DISCUSSION TITLE
Return back LIB folder

### DISCUSSION THREAD

**Discussion by markb-trustifi** [mozilla/pdf.js#18621] [Polls]
The "lib" folder existed in "pdfjs-dist" up to the version 3.5. After that is has been removed.
It contains very important image conversion utilities, for example "pdfjs-dist/lib/shared/image_utils".

**Comment by Snuffleupagus** [comment]
Please refer to PR #16350 to see why this was removed, hence it's unfortunately quite unlikely to be re-instated.

Note also that code found in e.g. `pdfjs-dist/lib/shared/image_utils` is internal functionality and thus *not* part of the public API, hence you'd not be able to rely on it being available in future versions anyway.

**Reply by markb-trustifi** [reply]
Then, instead of it you can export outside only this function "image_utils.convertToRGBA", or this one "CanvasGraphics.paintInlineImageXObject".

**Reply by Snuffleupagus** [reply]
Sorry, but neither of those are (at all) appropriate to expose in the official API since it's functionality that we must be able to modify freely.

---

In this particular case it might perhaps be easier to simply copy the relevant code into your implementation?
